projects/rero/ng-core/src/lib/record/search/aggregation/list-filters/list-filters.component.ts

```diff
@@ -55,19 +74,28 @@ export class ListFiltersComponent implements OnChanges {
       this._ref.detach();
       this.filters = [];
 
-      changes.aggregationsFilters.currentValue.map(filter => {
+      changes.aggregationsFilters.currentValue.map((filter: any) => {
         if (!this.filtersToHide.includes(filter.key)) {
           this.filters = this.filters.concat(
-            filter.values.map( value => {
-              return {key: value, aggregationKey: filter.key};
+            filter.values.map((value: any) => {
+              const data = { key: value, aggregationKey: filter.key };
+              // Check if value is a range of dates (string contain --)
+              if (value.indexOf('--') > 0) {
```

Comment:
I don't think checking "--" is the better idea to determine if the filter is a date interval.
I'm not a specialist but i think this string could appears in some specific rare case for author names, maybe keywords, ...
But I don't know a better solution to solve this problem at this time.

What is the content of `aggregationsFilters.currentValue` ? Do we have some information into this object about the relation between filter value and filter key ? 
Otherwise, it could be a little better to use a regexp to check the string. A date range filter should always be `/d[XX]--/d[XX]` (replace XX by number of digit)